### PR TITLE
odafileconverter: 21.7.0.0 -> 21.11.0.0

### PR DESCRIPTION
**Pull request by kinnala**
###### Motivation for this change

The current url is broken. Found the new url from https://www.opendesign.com/guestfiles/oda_file_converter. I built the new version successfully and converted some `dwg` files to `dxf` format.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/graphics/odafileconverter/default.nix [hunk 1]
```diff
@@ -14,8 +14,8 @@ in mkDerivation {
 
   src = fetchurl {
     # NB: this URL is not stable (i.e. the underlying file and the corresponding version will change over time)
-    url = "https://download.opendesign.com/guestfiles/ODAFileConverter/ODAFileConverter_QT5_lnxX64_7.2dll.deb";
-    sha256 = "0sa21nnwzqb6g7gl0z43smqgcd9h3xipj3cq2cl7ybfh3cvcxfi9";
+    url = "https://download.opendesign.com/guestfiles/Demo/ODAFileConverter_QT5_lnxX64_7.2dll_21.9.deb";
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/odafileconverter/default.nix:17]: 404 file not found.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/graphics/odafileconverter/default.nix (hunk 1 above)
Comment (kinnala) [reply pkgs/applications/graphics/odafileconverter/default.nix:17]: Seems that these URL's break very quickly; old version is deleted when a new version is released. What should we do? Have a mirror somehow?

Comment (SuperSandro2000) [reply pkgs/applications/graphics/odafileconverter/default.nix:17]: Some packages like this are mirrored on web.archive.org.

Comment (kinnala) [reply pkgs/applications/graphics/odafileconverter/default.nix]: Great idea. I did exactly that.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 101848` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>odafileconverter</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 101848` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)

**Pull request merged**